lmdeploy/cli/serve.py

```diff
@@ -135,6 +135,9 @@ def add_parser_api_server():
 
         # chat template args
         ArgumentHelper.chat_template(parser)
+        ArgumentHelper.tool_call_parser(parser)
+        ArgumentHelper.enable_reasoning(parser)
```

Comment:
we can use `ArgumentParser.add_mutually_exclusive_group` for `tool_call_parser` and `enable_reasoning`
https://docs.python.org/3/library/argparse.html#mutual-exclusion

Reply:
That is a good idea. But in the short term, they are mutually exclusive. I'm not sure if they work together in the future. It is not difficult to implement the feature.